src/Pact/Types/Gas.hs

```diff
@@ -150,14 +152,29 @@ data GasArgs
   -- ^ The cost of the in-memory representation of the module
   | GPrincipal !Int
   -- ^ the cost of principal creation and validation
-  | GIntegerOpCost !(Integer, Maybe Integer) !(Integer, Maybe Integer)
+  | GIntegerOpCost !(Integer, Maybe Integer) !(Integer, Maybe Integer) IntThreshold
   -- ^ Integer costs
   | GDecimalOpCost !Decimal !Decimal
   -- ^ Decimal costs
   | GMakeList2 !Integer !(Maybe Integer)
   -- ^ List versioning 2
   | GZKArgs ZKArg
 
+data IntThreshold
+  = ThresholdPrePact47
+  | ThresholdPact47
+  deriving (Show, Enum)
+
+thresholdToInteger :: IntThreshold -> Integer
+thresholdToInteger = \case
+  ThresholdPrePact47 -> (10 :: Integer) ^ (30 :: Integer)
```

Comment:
Can we comment on the rational\origins of these constants? They are unclear to me.

Reply:
10^30 was an arbitrary constant from before.
10^80 is just a close number to 2^256. Essentially, allowing for cheap math on numbers below that threshold.

Reply:
:smiley:
Thank you @jmcardon 

https://github.com/kadena-io/pact/issues/1152

Reply:
Yes, you were the motivation for increasing this limit, @CryptoPascal31 :)